pilot/pkg/serviceregistry/kube/controller/ambient/workloads.go

```diff
@@ -467,3 +486,27 @@ func getWorkloadEntryLocality(p *networkingv1alpha3.WorkloadEntry) *workloadapi.
 		Subzone: subzone,
 	}
 }
+
+func implicitWaypointPolicies(ctx krt.HandlerContext, Waypoints krt.Collection[Waypoint], waypoint *Waypoint, services []model.ServiceInfo) []string {
```

Comment:
TODO this results in:

* ALLOW  [workload waypoint's ids]
* ALLOW [service waypoints' ids]

This means the workloads that have both can't share a policy 1:1 with a Waypoint and need a per-workload policy to be generated. 

Or we could say that if there is a workload waypoint, enforce that over the service waypoint when there are both (most of the time they'll end up being the same wayp;oint e.g. namespace attached)